### PR TITLE
link-add: support hardware address

### PR DESCRIPTION
**Pull request by mie00**

### CODE REVIEW

**Comment by vishvananda** [comment]
Have you verified that this actually works? The iproute2 code only modifies the hwadress via a set call.

**Comment by squeed** [comment]
I wrote the same code (including tests!) and it worked for me - see https://github.com/vishvananda/netlink/compare/master...squeed:create-set-mac

**Comment by squeed** [comment]
Also, from what I can tell, the current version of iproute2 *does* set the mac address on create; see https://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git/tree/ip/iplink.c#n588

Of course, this isn't the easiest code to grok.

**Comment by vishvananda** [comment]
Hmm I see that you are correct, it is called in the create path as well. I will merge this. @mie00 can you rebase?

File: link_linux.go [hunk 1]
```diff
@@ -672,7 +672,7 @@ func LinkAdd(link Link) error {
 // are taken fromt the parameters in the link object.
 // Equivalent to: `ip link add $link`
 func (h *Handle) LinkAdd(link Link) error {
-	// TODO: set mtu and hardware address
+	// TODO: set mtu
```
Comment (squeed) [inline link_linux.go]: I believe MTU is set correctly now too, so this comment can be deleted.

**Comment by mie00** [comment]
I am sorry for my late response, I updated the PR.

**Comment by squeed** [comment]
Can you delete all of line 675? That comment is no longer correct.

**Comment by mie00** [comment]
Updated.

**Pull request merged**